src/parsec.rs

```diff
@@ -89,11 +133,19 @@ impl ParsecMap {
         pub_id: id::PublicId,
         log_ident: &LogIdent,
     ) -> bool {
-        let parsec = if let Some(parsec) = self.map.get_mut(&msg_version) {
-            parsec
-        } else {
-            return false;
-        };
+        let (parsec, size_counter) =
+            if let Some((parsec, size_counter)) = self.map.get_mut(&msg_version) {
+                (parsec, size_counter)
+            } else {
+                return false;
+            };
+
+        size_counter.increase_size(serialisation::serialised_size(&response));
```

Comment:
Are we processing the response again akin to a serialization to get the size?
If so, maybe that is not ideal.